packages/twenty-server/src/modules/connected-account/refresh-tokens-manager/services/connected-account-refresh-tokens.service.spec.ts

```diff
@@ -0,0 +1,148 @@
+import { Test, type TestingModule } from '@nestjs/testing';
+
+import { ConnectedAccountProvider } from 'twenty-shared/types';
+
+import { JwtWrapperService } from 'src/engine/core-modules/jwt/services/jwt-wrapper.service';
+import { TwentyORMManager } from 'src/engine/twenty-orm/twenty-orm.manager';
+import { MicrosoftAPIRefreshAccessTokenService } from 'src/modules/connected-account/refresh-tokens-manager/drivers/microsoft/services/microsoft-api-refresh-tokens.service';
+import {
+  ConnectedAccountRefreshAccessTokenException,
+  ConnectedAccountRefreshAccessTokenExceptionCode,
+} from 'src/modules/connected-account/refresh-tokens-manager/exceptions/connected-account-refresh-tokens.exception';
+import { type ConnectedAccountWorkspaceEntity } from 'src/modules/connected-account/standard-objects/connected-account.workspace-entity';
+
+import { ConnectedAccountRefreshTokensService } from './connected-account-refresh-tokens.service';
+
+describe('ConnectedAccountRefreshTokensService', () => {
+  let service: ConnectedAccountRefreshTokensService;
+  let microsoftAPIRefreshAccessTokenService: MicrosoftAPIRefreshAccessTokenService;
+  let twentyORMManager: TwentyORMManager;
+
+  const mockWorkspaceId = 'workspace-123';
+  const mockConnectedAccountId = 'account-456';
+  const mockAccessToken = 'valid-access-token';
+  const mockRefreshToken = 'valid-refresh-token';
+  const mockNewAccessToken = 'new-access-token';
+
+  beforeEach(async () => {
```

Comment:
to be honest, we should not invest on unit testing these services but rather on integration testing them. We should mock the low level microsoft/google clients so they return realistic API response and check that result is right in DB then

Reply:
(but that's fine for this PR, just a general note)